src/healthcheck.ts

```diff
@@ -4,17 +4,24 @@ import http from 'http';
 const app = express();
 const router = express.Router();
 
-router.use((req, res, next) => {
+router.use((req: express.Request, res: express.Response, next: express.NextFunction) => {
 	res.header('Access-Control-Allow-Methods', 'GET');
-	next();
+	const status = app.locals.getStatus()
+	if (status === 200) {
+		next();
+	} else {
+		res.status(status).send({ error: 'Something failed!' })
+	}
 });
```

Comment:
So it looks to me like you're doing the health check at a global middleware level rather than the health endpoint. Is that intentional? It would cause errors to be returned from every endpoint if the status check fails.